### ISSUE TITLE
Issue Rendering CardTable.java

### ISSUE THREAD

**Issue by nic-instruction** [ChrisMayfield/ThinkJavaCode2#4]
Hello,

This is my first time teaching from your textbook, and I've enjoyed it a lot, but I'm running into a problem when running CardTable.java in Eclipse.  It has no visible errors, but launches in the upper right-hand corner of the screen completely collapsed and doesn't render a deck when it is expanded.

From the Chapter 12 instructions I got the impression it would work out of the box and could then be used to implement a solitaire game.  I've attached a picture for reference.

The example graphics code in the appendix runs without issue and I haven't had trouble implementing other assignments as examples.  We were hoping to use this code as the basis for a class final project, so if there's a quick fix, let me know.

Thank you!
<img width="153" alt="Screen Shot 2021-03-09 at 10 01 31 PM" src="https://user-images.githubusercontent.com/22084241/110583860-2fc1ca80-8123-11eb-9e42-b95ef9ff03e1.png">

**Comment by ChrisMayfield** [comment]
I can't seem to reproduce the issue on my end. I tried running it from Eclipse on both my Mac laptop and Linux desktop, and in both cases, it shows the Card Table window correctly. I did notice a difference in the code from what we did in Chapters 15-17. Will you try changing Line 87 to `frame.add(canvas);` and let me know if that makes a difference on your end?

**Comment by nic-instruction** [comment]
Hi Chris,
Thank you so much for getting back to me so quickly!  I tested with the new line and the behavior did not change, you had already put something similar on line 87 in the version I'm using:


        Canvas canvas = new CardTable();

        //frame.getContentPane().add(canvas);

        frame.add(canvas);  //suggested line


So, I tried it with frame.getContentPane().add(canvas); (original), with frame.add(canvas) in addition, and with frame.getContentPane().add(canvas) commented out and  frame.add(canvas) on the next line (the paste above is from that).

One thing I noticed on my end is that if I change the name of the folder where the pictures are kept, I do not get an error, in fact, the behavior is the same.  I added a println statement and checked the names, they appear correct, I also played with adding a ./ before the folder name (I'm on a mac), but got the same result.  It may be something completely different, but my thought is that it is not finding the pictures for whatever reason.  I don't know if this would be a path issue, or what.  I did clone directly from your repository to make sure the file structure was correct, and got the same result.

Two of my students were able to reproduce this issue as well.  I'm wondering if we're missing something environmental?

Thank you!
________________________________

**Comment by ChrisMayfield** [comment]
You're probably right. I changed the name of the folder and got the same result (no error message and an empty window in the upper left corner). I looked at the source code for ImageIcon, and it indeed ignores FileNotFoundException. As a result, the height and width of each image object are -1. Nice work, Swing. :) I don't know if it would help, but I'm using OpenJDK 11 on both of my machines. It might be a bug in the JDK if you're using a different version.

**Comment by nic-instruction** [comment]
I'll try installing OpenJDK 11, and see if that helps.  I've tried moving it to a resources folder and changing cardset to resources on line 24, but that was not successful.  Hoping OpenJDK 11 will do the trick.  I'll let you know!


 String cardset = "resources";

________________________________

**Comment by nic-instruction** [comment]
Unfortunately I got the same result with OpenJDK 11.  (Picture attached).  
I did realize that I'm on a fairly new version of eclipse:

Version: 2020-09 (4.17.0)
Build id: 20200910-1200

And my students are a version after me, so I went down the class path rabbit hole.  For whatever reason it looks like the path only extends into the ch12 folder, and doesn't make it to the cardset-oxymoron folder at all, so I moved the folder out to the same level as the chapter folders, and then went into my project properties > Java Build Path > Source and added cardset-oxymoron to the source path.  This appears to have worked!  

Even being at the top level, the folder was not detected until I added it to the class path, so I don't know what the simple long-term solution to this problem is, but at least it's working!

Thank you so much for getting back to me so quickly, and for testing everything on your end, I really appreciate it!
<img width="790" alt="Screen Shot 2021-03-10 at 1 49 17 PM" src="https://user-images.githubusercontent.com/22084241/110705648-8d015e80-81ab-11eb-8863-b9a87042656e.png">
<img width="1143" alt="Screen Shot 2021-03-10 at 2 18 31 PM" src="https://user-images.githubusercontent.com/22084241/110705681-98548a00-81ab-11eb-8bff-de0b429daf05.png">
